### ISSUE TITLE
SPIKE: Bulk export from DataTables

### ISSUE THREAD

**Issue by mikeapp** [yalelibrary/YUL-DC#1441]
**Story**

The Beinecke DSU has requested a way to search/filter and export data to a CSV/Excel.   There appears to be an extension to DataTables that supports this:  https://datatables.net/extensions/buttons/

**Acceptance**
- [ ]  Attempt to add the export button to the existing Parent Object datatable (or all datatables).

**Comment by martinlovell** [comment]
We use the "buttons" plugin for the colvis (column visibility), and a custom button for the Clear Filters.
I spent a couple minutes looking at it: I tried just added "excel" to the list.  (and tried a few other things to quickly try to get it to show up)...but was unsuccessful.  It maybe that I wasn't importing all the necessary JS to get the button to appear and work.

**Comment by orangewolf** [comment]
After quite a few trials and tribulations I finally have both the excel and csv buttons loading and working. They also work with the column selection and filtering! https://share.getcloudapp.com/kpuD0wOj 

Questions:
- Do we want CSV button? (currently on, but easy to turn off)
- Do we want the PDF option (would require pdfjs to be added to the dependencies)
- Specs on this are a little light, I followed the example of colvis and simply added "these buttons show up" tests. There isn't anything to test that they work or how. Having said that, I submit that as ok simply because getting them "on" was the hard part. They just worked slick after that and specs around how they function would be more testing the upstream Datatables library than testing our application code, as our app code just turns them on.

Some notes:
- it turns out the though much of the documentation calls for "excel" it is actually "excelHtml5" that we want and is currently supported.  This https://datatables.net/extensions/buttons/examples/html5/simple.html was the page that finally clued me in to that.
- I was originally way too focused on the Excel button. I had just about given up on it when I tried to able to get CSV to work as a last ditch and that helped me understand what was wrong with Excel.
- If jszip is not loaded properly, the Excel button simply does not show up. There is no error or log message around why. I had to decompile the code and put in debug statements to figure this out.
